ot/lowrank.py

```diff
@@ -0,0 +1,502 @@
+    - :math: `\mathcal{C(a,b,r)}` are the low-rank couplings of the OT problem
+
+
+    Parameters
+    ----------
+    X_s : array-like, shape (n_samples_a, dim)
+        samples in the source domain
+    X_t : array-like, shape (n_samples_b, dim)
+        samples in the target domain
+    a : array-like, shape (n_samples_a,)
+        samples weights in the source domain
+    b : array-like, shape (n_samples_b,)
+        samples weights in the target domain
+    reg : float, optional
+        Regularization term >0
+    rank : int, optional. Default is None. (>0)
+        Nonnegative rank of the OT plan. If None, min(ns, nt) is considered.
+    alpha : int, optional. Default is None. (>0 and <1/r)
+        Lower bound for the weight vector g. If None, 1e-10 is considered
+    rescale_cost : bool, optional. Default is False
+        Rescale the low rank factorization of the sqeuclidean cost matrix
+    init : str, optional. Default is None. If None, "random" is considered.
+        Initialization strategy for the low rank couplings. 'random', 'trivial' or 'kmeans'
+    reg_init : float, optional. Default is None. (>0)
+        Regularization term for a 'kmeans' init. If None, 1 is considered.
+    seed_init : int, optional. Default is None. (>0)
+        Random state for a 'random' or 'kmeans' init strategy.
+    gamma_init : str, optional. If None, 'rescale' is considered
+        Initialization strategy for gamma. 'rescale', or 'theory'
+    numItermax : int, optional. Default is 2000.
+        Max number of iterations for the Dykstra algorithm
+    stopThr : float, optional. Default is 1e-7.
+        Stop threshold on error (>0) in Dykstra
+    warn : bool, optional
+        if True, raises a warning if the algorithm doesn't convergence.
+    log : bool, optional
+        record log if True
+
+
+    Returns
+    ---------
+    Q : array-like, shape (n_samples_a, r)
+        First low-rank matrix decomposition of the OT plan
+    R: array-like, shape (n_samples_b, r)
+        Second low-rank matrix decomposition of the OT plan
+    g : array-like, shape (r, )
+        Weight vector for the low-rank decomposition of the OT
+    log : dict (lazy_plan, value and value_linear)
+        log dictionary return only if log==True in parameters
+
+
+    References
+    ----------
+    .. [65] Scetbon, M., Cuturi, M., & Peyré, G. (2021).
+        "Low-rank Sinkhorn Factorization". In International Conference on Machine Learning.
+
+    """
+
+    # POT backend
+    nx = get_backend(X_s, X_t)
+    ns, nt = X_s.shape[0], X_t.shape[0]
+
+    # Initialize weights a, b
+    if a is None:
+        a = unif(ns, type_as=X_s)
+    if b is None:
+        b = unif(nt, type_as=X_t)
+
+    # Compute rank (see Section 3.1, def 1)
+    r = rank
+    if rank is None:
+        r = min(ns, nt)
+    else:
+        r = min(ns, nt, rank)
+
+    if r <= 0:
+        raise ValueError("The rank parameter cannot have a negative value")
+
+    # Compute alpha
```

Comment:
It will be more explicit for the user to set directly `alpha=1e-10` as default value in the function parameter, and adapt the doc accordingly.

Reply:
I'll change the default value for alpha, init and gamma_init